Core/Node-API/Include/Shared/napi/napi.h

```diff
@@ -36,6 +36,7 @@
 #endif  // NAPI_HAS_THREADS
 #include <string>
 #include <vector>
+#include <exception>
```

Comment:
Should this be conditional behind NAPI_CPP_EXCEPTIONS?

Reply:
Yes, it should. We also need to guard both ```static Error New(napi_env env, const std::exception& exception)``` and ```static Error New(napi_env env, const std::exception_ptr& exception_ptr)``` using NAPI_CPP_EXCEPTIONS  in the napi.h. We are already doing that for the implementation in the cpp file but I think we forgot to do it for the header.